src/logging.c

```diff
@@ -0,0 +1,55 @@
+#include <stdarg.h>
+#include <time.h>
+
+#include "logging.h"
+
+
+/******************************************************
+ *** LOCAL DEFINES                                  ***
+ ******************************************************/
+
+/******************************************************
+ *** LOCAL VARIABLES                                ***
+ ******************************************************/
+
+static LoggingLevel min_dbg_lvl = FATAL;
+
+/******************************************************
+ *** LOCAL METHODS                                  ***
+ ******************************************************/
+
+/******************************************************
+ *** EXPOSED METHODS                                ***
+ ******************************************************/
+
+void log_set_lvl(const CLIArguments* conf) {
+    min_dbg_lvl = (LoggingLevel) conf->debug;
+
+    // minimum debug level cannot be higher than fatal
+    if (min_dbg_lvl > FATAL)
+        min_dbg_lvl = FATAL;
+
+    // minimum debug level cannot be lower than debug
+    if (min_dbg_lvl < DEBUG)
+        min_dbg_lvl = DEBUG;
+
+    LOG_INFO("Log level initialized to %d.", min_dbg_lvl);
```

Comment:
Maybe we should print the name of the enum member, instead of the int value?

Reply:
Ah nvm, the cli takes a number too 🤔

Reply:
Printing the name of the log level would require to map the int to a string just for that, I don't think that this is really worth it